core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingUtils.java

```diff
@@ -18,4 +20,25 @@ public static Path makePath(BlockInfra blockInfra, RawSignalingInfra infra, Inte
         }
         return buildPathFrom(infra, chunks, 0, totalLength);
     }
+
+
+    /**
+     * @param infra FullInfra
+     * @param blockIds the blocks in the order they are encountered
+     * @param offsetFirstBlock the path offset on the first block in millimeters
+     * @return corresponding path
+     */
+    public static Path makePathFromBlocks(FullInfra infra,
+                                          List<Integer> blockIds,
+                                          long offsetFirstBlock) {
+        var chunks = new MutableDirStaticIdxArrayList<TrackChunk>();
+        var totalLength = 0;
```

Comment:
```suggestion
        long totalLength = 0;
```

Otherwise it would be an `int`